File: langfood-backend/LangFood.Shared/Models/Building.cs
using System.ComponentModel.DataAnnotations;

namespace LangFood.Shared.Models
{
    public class Building
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Tên tòa nhà không được để trống")]
        [StringLength(100)]
        public string Name { get; set; }

        public bool IsActive { get; set; } = true;

        // Nếu muốn lưu thêm mô tả hoặc vị trí bản đồ có thể thêm ở đây
        // public string? Description { get; set; }
    }
}

File: langfood-android/app/src/main/java/com/example/langfood/models/Building.java
package com.example.langfood.models;

import com.google.gson.annotations.SerializedName;
import java.io.Serializable;

public class Building implements Serializable {
    @SerializedName(value = "id", alternate = {"Id"})
    private int id;

    @SerializedName(value = "name", alternate = {"Name"})
    private String name;

    @SerializedName(value = "isActive", alternate = {"IsActive"})
    private boolean isActive;

    public Building() {}

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public boolean isActive() { return isActive; }
    public void setActive(boolean active) { isActive = active; }

    @Override
    public String toString() {
        return name; // Để hiển thị trong Spinner nếu cần
    }
}
